lib/manager/terraform/modules.ts

```diff
@@ -62,6 +63,10 @@ export function analyseTerraformModule(dep: PackageDependency): void {
     if (moduleParts[0] === '..') {
       dep.skipReason = SkipReason.Local;
     } else if (moduleParts.length >= 3) {
+      const hostnameMatch = hostnameMatchRegex.exec(dep.managerData.source);
+      if (hostnameMatch) {
+        dep.registryUrls = [hostnameMatch.groups.hostname];
+      }
```

Comment:
This assumes that `https` is always used? I think a better secure solution is to use `URL.parse` and add `protocol`, `domainName` and `port` to resulting `registryUrl`.

Reply:
thx, acutally http(s) can be dropped in the regex. 
I can't use URL here to parse this string `app.terraform.io/foo/bar` as well as `hashicorp/consul/aws`. The later would contain only `hashicorp`  which suggests that this is a public module. 

Regarding protocol and port, it seems like Hashicorp is enforcing to HTTPS and port 443. 

> A user-facing hostname is a fully-specified internationalized domain name expressed in its Unicode form (the corresponding "punycode" form is not allowed) which must be resolvable in DNS to an address that has an HTTPS server running on port 443.

https://www.terraform.io/docs/internals/remote-service-discovery.html#user-facing-hostname

Reply:
Ok, then it's fine. 👍 Thanks for explaining.